internal/observability/tracing/tracing.go

```diff
@@ -83,14 +85,27 @@ func configureJaeger(ctx context.Context) error {
 }
 
 func configureOTLP(ctx context.Context) error {
-	conn, err := grpc.DialContext(ctx, conf.OTLP.CollectorEndpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
-	if err != nil {
-		return fmt.Errorf("failed to dial otlp collector: %w", err)
-	}
+	var exporter *otlptrace.Exporter
+	var err error
 
-	exporter, err := otlp.New(ctx, otlp.WithGRPCConn(conn))
-	if err != nil {
-		return fmt.Errorf("failed to create otlp exporter: %w", err)
+	switch conf.OTLP.Protocol {
+	case "grpc":
+		conn, err := grpc.DialContext(ctx, conf.OTLP.CollectorEndpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			return fmt.Errorf("failed to dial otlp collector: %w", err)
+		}
+
+		exporter, err = otlp.New(ctx, otlp.WithGRPCConn(conn))
+		if err != nil {
+			return fmt.Errorf("failed to create otlp exporter: %w", err)
+		}
+	case "http":
+		exporter, err = otlphttp.New(ctx, otlphttp.WithEndpoint(conf.OTLP.CollectorEndpoint))
+		if err != nil {
+			return fmt.Errorf("failed to create otlp exporter: %w", err)
+		}
```

Comment:
maybe we can add it to another pull request for this feature so that the pr additions can be better understood, so after I merge this pull request, I will open an issue and continue my improvements on it